rules/sort-array-includes.ts

```diff
@@ -218,11 +221,13 @@ export let sortArray = <MessageIds extends string>({
         selector = 'literal'
       }
 
-      for (let predefinedGroup of generatePredefinedGroups({
+      let predefinedGroups = generatePredefinedGroups({
```

Comment:
[nothing to change] 

> Move the `generatePredefinedGroups` function call outside the loop. Currently, this function is called at each iteration of the loop.

Performance wise, the previous and new code is the same:

```ts
for (let predefinedGroup of generatePredefinedGroups({ 
``` 

will not call `generatePredefinedGroups` at each iteration. 🙂

I agree this is cleaner to write it your way though!